src/Microsoft.DotNet.Interactive.PackageManagement/PackageRestoreContext.cs

```diff
@@ -51,7 +54,10 @@ private IEnumerable<string> NativeProbingRoots()
             }
         }
 
-        public void AddRestoreSource(string source) => _restoreSources.Add(source);
+        public void AddRestoreSource(string source)
```

Comment:
Given then change in the implementation, should this be `TryAddRestoreSource`?

Reply:
There is not really a failure mode, it either adds the source or it was already there.  So Try is not really valuable on the name ... but GetOrAdd is indeed a bit clunky, however, it's the form I have seen most commonly ...

Reply:
Got to say GetOrAdd on ConcurrentDictionary is a bit tricky to parse too.

Reply:
Not critical. I do think `Try*` helps message the intention a bit better.

Reply:
The intention is to add the restore, whether it is in or not.  It's not to Try and add the restore, but sure I can make the change since you think it makes it clearer.

Reply:
I think it only makes it clearer because a bunch of other APIs with this behavior use the same naming convention. If I _really_ wanted to make it clearer, I'd call it something like `AddRestoreSourceIfSomeoneDidntAlreadyAddItAndDontThrowIfTheyDid`. 😂

Reply:
> Got to say GetOrAdd on ConcurrentDictionary is a bit tricky to parse too.

Yeah, it seems to imply that if it's going to be added, then you can't get it. _**It's one or the other!**_

Reply:
````
AddRestoreSourceIfSomeoneDidntAlreadyAddItAndDontThrowIfTheyDid
````
Now that is a proper DotNet Api name.